File: frontend/lib/models/conversation.dart
class Conversation {
  final int id;
  final String conversationType; // voice, video, text
  final String? participantType; // reed, bootie_boss, admin, player
  final String? contextSummary;
  final DateTime? lastMessageAt;
  final DateTime createdAt;

  Conversation({
    required this.id,
    required this.conversationType,
    this.participantType,
    this.contextSummary,
    this.lastMessageAt,
    required this.createdAt,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) {
    return Conversation(
      id: json['id'] as int,
      conversationType: json['conversation_type'] as String,
      participantType: json['participant_type'] as String?,
      contextSummary: json['context_summary'] as String?,
      lastMessageAt: json['last_message_at'] != null
          ? DateTime.parse(json['last_message_at'] as String)
          : null,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  bool get isVoice => conversationType == 'voice';
  bool get isVideo => conversationType == 'video';
  bool get isText => conversationType == 'text';
  bool get isReed => participantType == 'reed';
}

File: app/models/conversation.rb
class Conversation < FirestoreModel
  attribute :user_id, :string
  attribute :participant_user_id, :string
  attribute :conversation_type, :string
  attribute :participant_type, :string
  attribute :context_summary, :string
  attribute :last_message_at, :datetime

  validates :conversation_type, presence: true, inclusion: { in: %w[voice video text] }
  validates :participant_type, inclusion: { in: %w[reed bootie_boss admin player] }, allow_nil: true

  def self.recent
    order(:last_message_at, :desc).get
  end

  def self.unread
    # Get conversations with unread messages
    unread_message_conversation_ids = Message.where(:read, false).get.map(&:conversation_id).uniq
    return [] if unread_message_conversation_ids.empty?
    unread_message_conversation_ids.map { |id| find(id) }.compact
  end

  def update_last_message!
    update(last_message_at: Time.current)
  end

  def user
    User.find(user_id) if user_id.present?
  end

  def participant_user
    User.find(participant_user_id) if participant_user_id.present?
  end

  def messages
    Message.where(:conversation_id, id).get
  end
end
